### PR TITLE
fix(sinks): set fixed buffer size for distributed service

### PR DESCRIPTION
**Pull request by dsmith3197**
This buffer was added in https://github.com/vectordotdev/vector/pull/13918. With its current capacity, it can silently buffer `200 * # of endpoints` requests, which can result in substantial memory growth. After examining the code, it seems that this buffer only needs a size of 1.

This currently impacts the elasticsearch sink.

### CODE REVIEW

File: src/sinks/util/service.rs [hunk 1]
```diff
@@ -390,7 +388,6 @@ impl TowerRequestSettings {
         ServiceBuilder::new()
             .rate_limit(self.rate_limit_num, self.rate_limit_duration)
             .retry(policy)
-            .layer(BufferLayer::new(max_concurrency))
```
Comment (dsmith3197) [inline src/sinks/util/service.rs:393]: This is actually included so that the service implements `Clone`.

**Review by neuronull — COMMENTED** [review]
Are the ES int tests added in https://github.com/vectordotdev/vector/pull/14088 sufficient to regression test this?

**Comment by dsmith3197** [comment]
> > Are the ES int tests added in #14088 sufficient to regression test this?
> 
> No, I don't believe we have any regression cases that would cover this. I can look into adding one.

@neuronull I added a suitable regression test and no change was detected.

**Review by neuronull — APPROVED** [review]

**Review by jszwedko — COMMENTED** [review]
Nice find and 👍 on the new regression test.

**Pull request merged**